frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx

```diff
@@ -241,27 +241,49 @@ const getNodesForTasks = (
   return [nodes, children];
 };
 
+type UsePipelineTaskTopology = {
+  nodes: PipelineNodeModelExpanded[];
+  loaded: boolean;
+  error: Error | undefined;
+};
+
 export const usePipelineTaskTopology = (
   spec?: PipelineSpecVariable,
   runDetails?: RunDetailsKF,
   executions?: Execution[] | null,
   events?: Event[] | null,
   artifacts?: Artifact[],
-): PipelineNodeModelExpanded[] =>
-  React.useMemo(() => {
+): UsePipelineTaskTopology => {
+  const [nodes, setNodes] = React.useState<PipelineNodeModelExpanded[]>([]);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<Error | undefined>(undefined);
+
+  React.useEffect(() => {
     if (!spec) {
-      return [];
+      setNodes([]);
+      setLoaded(true);
+      setError(new Error('No spec found'));
+      return;
     }
     const pipelineSpec = spec.pipeline_spec ?? spec;
 
-    const {
-      components,
-      deploymentSpec: { executors },
-      root: {
-        dag: { tasks },
-        inputDefinitions,
-      },
-    } = pipelineSpec;
+    let components, executors, tasks, inputDefinitions;
+
+    try {
+      ({
+        components,
+        deploymentSpec: { executors },
+        root: {
+          dag: { tasks },
+          inputDefinitions,
+        },
+      } = pipelineSpec);
+    } catch (e) {
+      setNodes([]);
+      setLoaded(true);
+      setError(new Error('Error processing pipelineSpec'));
+      return;
```

Comment:
```
components,
        deploymentSpec: { executors },
        root: {
          dag: { tasks },
          inputDefinitions,
        },
      } = pipelineSpec
```
This will throw the error with argo workflow, as it doesn't have inputDefinitions and other.